applications/app/templates/serviceWorker.scala.js

```diff
@@ -150,6 +150,24 @@ var requestForAsset = (function () {
     }
 })();
 
+// handle requests for generic pages
+var requestForPage = function (event) {
```

Comment:
hmmm, that's a good point. currently, it's work as you'd expect, but I think you're right – that must be a race condition?